crates/core/tedge_mapper/src/c8y/mapper.rs

```diff
@@ -193,7 +197,7 @@ pub async fn create_mqtt_client_will_message(
             "down",
             "service",
             None,
-        ));
+        ).with_retain());
```

Comment:
The `.with_retain()` has to be added by the `service_monitor_status_message` function and not by all and every callers.

Reply:
The connection created by `create_mqtt_client_will_message`  must also define an init message to be send on reconnect.